pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -55,6 +55,11 @@ const (
 	// leaking a blob that is no longer tagged in given repository.
 	BlobRepositoryCacheTTLEnvVar = "REGISTRY_MIDDLEWARE_REPOSITORY_OPENSHIFT_BLOBREPOSITORYCACHETTL"
 
+	// PullthroughInsecureEnvVar is a boolean environment variable that allows to get image from the remote
+	// registry using HTTP protocol. This option requires that the pullthrough is enabled.
```

Comment:
Insecure flag allows for pulling not only over `HTTP` protocol but also over `HTTPS` with self-signed certificate or with generally over untrusted connection.

I like a description of `--insecure` flag of `image-import` command:

> allow importing from registries that have invalid HTTPS certificates or are hosted via HTTP.